### PR TITLE
Feat DTMT 14: give tags to files in config

### PR DESCRIPTION
**Pull request by Lysoun**

### CODE REVIEW

File: modules/cli/src/main/kotlin/datamaintain/cli/App.kt [hunk 1]
```diff
@@ -40,6 +42,19 @@ class App : CliktCommand() {
 
     private val mongoTmpPath: String? by option(help = "mongo tmp file path")
 
+    private val tags: Set<Tag>? by option(help = "list of your tags defined using glob path matchers, like this: " +
+            "MYTAG1=[pathMatcher1; pathMatcher2],MYTAG2=[pathMatcher3]...")
+            .convert {
+                it.split(",")
```
Comment (driccio) [inline modules/cli/src/main/kotlin/datamaintain/cli/App.kt]: All this block is duplicated with DatamaintainConfig. Can't we find a way to have this code in one place ?

File: modules/core/src/main/kotlin/datamaintain/core/step/Scanner.kt [hunk 2]
```diff
@@ -13,16 +14,20 @@ class Scanner(private val context: Context) {
...
-                        tags.addAll(buildTagsFromFolder(rootFolder, it));
-                    }
+    private fun buildTags(config: DatamaintainConfig, rootFolder: File, file: File): Set<Tag> {
+        val tags = mutableSetOf<Tag>()
 
-                    FileScript(it.toPath(), context.config.identifierRegex, tags.toSet())
-                }
-                .sortedBy { it.name }.toList()
+        if(config.doesCreateTagsFromFolder)
+            tags.addAll(buildTagsFromFolder(rootFolder, file))
```
Comment (driccio) [inline modules/core/src/main/kotlin/datamaintain/core/step/Scanner.kt]: I like space between `if` and `(`. I prefer to have clearly identified block `if` using `{` and `}` event if this block contains only one line.

File: modules/core/src/test/kotlin/datamaintain/core/DatamaintainConfigTest.kt [hunk 3]
```diff
@@ -26,6 +28,14 @@ class DatamaintainConfigTest {
             get { blacklistedTags }.isEqualTo(setOf(Tag("un"), Tag("deux")))
             get { doesCreateTagsFromFolder }.isTrue()
             get { executionMode }.isEqualTo(ExecutionMode.DRY)
+            get { tags }.map { it.name }.containsExactly("TOTO", "potato")
+//                    setOf(Tag("TOTO", setOf(
```
Comment (driccio) [inline modules/core/src/test/kotlin/datamaintain/core/DatamaintainConfigTest.kt]: Why so many code commented  ?

File: modules/core/src/test/resources/config/default.properties [hunk 4]
```diff
@@ -2,5 +2,6 @@ scan.path=/tmp/test
 scan.identifier.regex=(.*?)_.*
 scan.tags.createFromFolder=true
 filter.tags.blacklisted=un,deux
+tags=TOTO=[src/test/resources/scanner_test_files/01_file1;src/test/resources/scanner_test_files/subfolder/*],potato=[src/test/resources/scanner_test_files/*;src/test/resources/scanner_test_files/subfolder/03_file3]
```
Comment (driccio) [inline modules/core/src/test/resources/config/default.properties]: We didn't speak about the way to configure tags. I guess we should :-)

I would prefer :
- one property line per tag
- glob path matchers list with comma as separator

Example:
`
tags.TOTO=[src/test/resources/scanner_test_files/01_file1,src/test/resources/scanner_test_files/subfolder/*]
tags.potato=[src/test/resources/scanner_test_files/*,src/test/resources/scanner_test_files/subfolder/03_file3]
`

Don't you find it is more readable ?
Same remark for the CLI. `--tag TOTO=[] --tag potato=[]`

Free to speak with you about that.

File: modules/core/src/main/kotlin/datamaintain/core/step/Scanner.kt (hunk 2 above)
Comment (Lysoun) [reply modules/core/src/main/kotlin/datamaintain/core/step/Scanner.kt]: Alright, I've added curly brackets in commit e7b6a570a7e5f7d0d26a7a70c782eb647eb358c9

**Comment by Lysoun** [comment]
We're working on it with @driccio, the code currently present in `Tag` is going to be split between two classes: 
- `Tag`, which keeps `name`, `matchedBy` and `matches`
- `TagMatcher` which will handle the path matching, exposing a method `matches` and using path matchers created from a list of string to match, or not, the path given. It also contains a `Tag`.
Refactoring will come during the next work session.

**Pull request merged**